includes/WikiDiscover.php

```diff
@@ -130,19 +130,53 @@ public static function onParserGetVariableValueSwitch(
 		&$cache,
 		$magicWordId,
 		&$ret ) {
+		$parser->setFunctionHook( 'numberofwikisincategory', [ __CLASS__, 'numberofwikisincategory' ], Parser::SFH_NO_HASH );
+		
 		if ( $magicWordId == 'numberofwikis' ) {
 			global $wgLocalDatabases;
 			$ret = $cache[$magicWordId] = count( $wgLocalDatabases );
 		}
+		
+		if ( $magicWordId == 'numberofprivatewikis' ) {
+			global $wgCreateWikiDatabase;
+			$dbw = wfGetDB( DB_MASTER, [], $wgCreateWikiDatabase );
+			$ret = $cache[$magicWordId] = $dbw->selectRowCount( 'cw_wikis', '*', [ 'wiki_private' => 1 ] );
+		}
+		
+		if ( $magicWordId == 'numberofactivewikis' ) {
+			global $wgCreateWikiDatabase;
+			$dbw = wfGetDB( DB_MASTER, [], $wgCreateWikiDatabase );
+			$ret = $cache[$magicWordId] = $dbw->selectRowCount( 'cw_wikis', '*', [ 'wiki_inactive' => 0 ] );
+		}
+		
+		if ( $magicWordId == 'numberofclosedwikis' ) {
+			global $wgCreateWikiDatabase;
+			$dbw = wfGetDB( DB_MASTER, [], $wgCreateWikiDatabase );
+			$ret = $cache[$magicWordId] = $dbw->selectRowCount( 'cw_wikis', '*', [ 'wiki_closed' => 1 ] );
+		}
+		
 		return true;
 	}
 
+	public static function numberofwikisincategory( Parser $parser, $category = '' ) {
```

Comment:
camelCase + if we're defining the type for $parser, can we define the type of $category?

Reply:
Fixed